c/yaml2json.c

```diff
@@ -123,50 +123,72 @@ static int yamlReadHandler(void *data, unsigned char *buffer, size_t size, size_
 }
 
 static void decodeParserError(yaml_parser_t *parser, char *errorBuf, size_t errorBufSize) {
+  size_t problemLen = strlen(parser->problem);
+  char problemNative[problemLen + 1];
+  snprintf (problemNative, problemLen + 1, "%s", parser->problem);
+  convertToNative(problemNative, problemLen);
+  char *nativeContext = NULL;
+  if (parser->context) {
+    size_t contextLen = strlen(parser->context);
+    char contextNative[contextLen + 1];
+    snprintf(contextNative, contextLen + 1, "%s", parser->context);
+    convertToNative(contextNative, contextLen);
+    nativeContext = contextNative;
+  }
+  /*
+   * We aren't sending back the parser far enough to determine the type of error. We
+   * want to know the file in question, so we'll rely on the processor of this errorBuffer
+   * to use replaceString("&name", name).
+   */
   switch (parser->error) {
     case YAML_MEMORY_ERROR:
-      snprintf(errorBuf, errorBufSize, "YAML memory error: not enough memory for parsing");
+      snprintf(errorBuf, errorBufSize, "Couldn't allocate enough memory to process file '&name'.");
       break;
     case YAML_READER_ERROR: {
-      size_t problemLen = strlen(parser->problem);
-      char problemNative[problemLen + 1];
-      snprintf (problemNative, problemLen + 1, "%s", parser->problem);
-      convertToNative(problemNative, problemLen);
       if (parser->problem_value != -1) {
-        snprintf(errorBuf, errorBufSize, "YAML reader error: %s: #%X at %ld", problemNative, parser->problem_value, (long)parser->problem_offset);
+        snprintf(errorBuf,
+                 errorBufSize,
+                 "Couldn't read file '&name': %s, #%x at %ld.",
+                 problemNative, parser->problem_value, (long)parser->problem_offset);
```

Comment:
Is this `size_t`? Then you don't need to cast, just use `%zu`.

Reply:
I don't believe it is size_t.

Reply:
Sorry, I wasn't clear enough, I was talking about `problem_offset`. Is that this field: https://github.com/yaml/libyaml/blob/0.2.5/include/yaml.h#L1099 ?

Reply:
Yes. I see what you mean now. Thanks!